### ISSUE TITLE
astro-function only working till change of day

### ISSUE THREAD

**Issue by mcmuller** [ioBroker/ioBroker.javascript#31]
As discussed [http://forum.iobroker.net/viewtopic.php?f=21&t=4693](url) for me on my fresh Raspberry-Install (from provided image) 

```
schedule({astro: "sunrise"}, function () {
    log("Sunrise!");
});
```

only works once after java-instance is loaded. The second trigger time (no matter if sunrise, sunset,...) is not executed. `getAstroDate()` is working as expected.

...hope, I'm right to file this (mine) issue here.

**Comment by mcmuller** [comment]
Checked with javascript-adaper 3.3.8 and Host 1.1.2 - problem still exists.

**Comment by mcmuller** [comment]
now reinstalled iobroker on a plain Debian system  - issue still exists, at least for me. Would be great if someone can confirm if it's just for me or not.

**Comment by mcmuller** [comment]
further Observation: I changed the Long/Lat from my home-Position (37.10 -8.50) to two positive values (37.10 8.50) and voilá, it's working.

**Comment by GermanBluefox** [comment]
Could you check the github version now?
https://github.com/ioBroker/ioBroker.javascript/commit/b7d56ca447a5268daff96734ae870859e44c951f

**Comment by mcmuller** [comment]
module restart results in error:
host.spok | 2017-11-03 14:42:37.050 | error | instance system.adapter.javascript.0 terminated with code 1 ()
-- | -- | -- | --
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Function.Module._load (module.js:300:12)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Module.load (module.js:343:32)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Object.Module._extensions..js (module.js:416:10)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Module._compile (module.js:409:26)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Object. (/opt/iobroker/node_modules/iobroker.javascript/javascript.js:3307:3)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at /opt/iobroker/node_modules/iobroker.javascript/javascript.js:22:28
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at require (internal/module.js:12:17)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Module.require (module.js:353:17)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Function.Module._load (module.js:276:25)
Caught | 2017-11-03 14:42:37.050 | error | by controller[0]: at Function.Module._resolveFilename (module.js:325:15)
Caught | 2017-11-03 14:42:37.049 | error | by controller[0]: Error: Cannot find module 'coffee-compiler'
Caught | 2017-11-03 14:42:37.049 | error | by controller[0]: ^
Caught | 2017-11-03 14:42:37.049 | error | by controller[0]: throw err;
Caught | 2017-11-03 14:42:37.049 | error | by controller[0]: module.js:327
host.spok | 2017-11-03 14:42:36.920 | info | instance system.adapter.javascript.0 started with pid 24689

**Comment by GermanBluefox** [comment]
Your installation went wrong. Could you please fix it and try again?

2017-11-03 15:45 GMT+01:00 mcmuller <notifications@github.com>:

**Comment by mcmuller** [comment]
OK. installs fine now (with the GUI-help instead of git) Will see the next ~30h

**Comment by mcmuller** [comment]
Works fine till the first change of day, then triggers no more till adapter restart.

**Comment by PascalBru** [comment]
is there any progress on the issue?

**Comment by mcmuller** [comment]
...been awhile, but even after moving the install from raspberry to freenas, the problem still exists: My workaroud is to restart the java-adapter every night....

**Comment by Apollon77** [comment]
We figured out one thing: When you start JS Adapter between 0 and 1 at night then this is broken somehow. We still need to figure out what exactly happens there, we think it could be something with UTC or such.

As soon as you start JS Adapter before 0 or after 1 everything works well.

Can it be that you have a reboot/backup (with stop) or something in these timeframes?

**Comment by paul53** [comment]
It is fixed since Version 4.1.12.
